src/log.rs

```diff
@@ -28,15 +38,27 @@ pub fn show(mut connection: Connection, verbose: bool) {
                 } else {
                     Time::Local(Local::now().naive_local())
                 };
-                let qos = format::qos(publish.qos);
                 let topic = publish.topic;
                 let size = publish.payload.len();
                 let payload = Payload::unlimited(publish.payload.into());
-                println!("{time:12} QoS:{qos:11} {topic:50} Payload({size:>3}): {payload}");
+                if json {
+                    let output = serde_json::to_string(&JsonLog {
+                        time,
+                        qos: publish.qos as u8,
+                        topic,
+                        size,
+                        payload,
+                    })
+                    .expect("Should be able to format log line as JSON");
+                    println!("{output}");
```

Comment:
Another minor nitpick: what kind of output? Every return value of something is output. Its not perfect naming it JSON, but it seems better than output to me. Maybe you have another better idea?
```suggestion
                    let json = serde_json::to_string(&JsonLog {
                        time,
                        qos: publish.qos as u8,
                        topic,
                        size,
                        payload,
                    })
                    .expect("Should be able to format log line as JSON");
                    println!("{json}");
```